### PR TITLE
Better error message when specifying file as target for remove

### LINKED ISSUE #5288: remove: support file as a target
### Description
As a further improvement of #4497, the support of using file as a target for ```dvc remove``` command should be added.

Here is what @skshetry says (https://github.com/iterative/dvc/pull/5236#issuecomment-759330299):
> we need to support file as a target in the remove, so I am not sure if this new error message makes sense.
> 
> We do need to clean up the message, especially the "failed to remove" one. So, either we need to add support for it, or just do the cleanup in this PR.
> 
> Regarding the support for files, the change should be pretty trivial, we just need collect_granular here, similar to what's in the repo.commit:
> 
> **dvc/dvc/repo/commit.py** (Line 44 in 228b3b1):
> ``` stages_info = self.stage.collect_granular( )```

@efiop made another point (https://github.com/iterative/dvc/issues/4497#issuecomment-759298988) based on a user's feedback:
> Another user ran into it https://discord.com/channels/485586884165107732/485596304961962003/798834932217937970

/cc @efiop @skshetry

### PR DESCRIPTION
**Pull request by alexmojaki**
Fixes specifically https://github.com/iterative/dvc/issues/5288#issuecomment-1176844757

* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

### CODE REVIEW

File: dvc/repo/remove.py [hunk 1]
```diff
@@ -11,7 +18,25 @@
...
 def remove(self: "Repo", target: str, outs: bool = False):
-    stages = self.stage.from_target(target)
+    try:
+        stages = self.stage.from_target(target)
+    except (StageNotFound, StageFileDoesNotExistError) as e:
+        # If the user specified a tracked file as a target instead of a stage,
+        # e.g. `data.csv` instead of `data.csv.dvc`,
+        # give a more helpful error message.
+        try:
+            self.stage.from_target(target + DVC_FILE_SUFFIX)
```
Comment (daavoo) [inline dvc/repo/remove.py:28]: Since this only shows a hint, could we check if `target + DVC_FILE_SUFFIX` file exists?

Comment (alexmojaki) [reply dvc/repo/remove.py:28]: I think I have to leave this question to your team, I don't know dvc well enough. Maybe ping someone else to chime in. Can you find a concrete, plausible scenario where it makes a difference, and explain why the result of either approach is better?

Comment (efiop) [reply dvc/repo/remove.py:28]: I agree with @daavoo , we should just check if it exists here. Or not check at all even and just show the hint (we do the same in checkout).

**Comment by alexmojaki** [comment]
The test failure is an unrelated celery timeout.

**Review by efiop — APPROVED** [review]
Thank you!

**Pull request merged**